File: Java/A_Star/src/A_star/Spot.java
package A_star;

import java.util.ArrayList;

public class Spot {

    protected Spot(int value, int x, int y) {
        this.value = value;
        this.x = x;
        this.y = y;
        this.h = 0;
        this.g = 0;
        this.f = 0;
        this.previous = null;
        this.neighbors = new ArrayList<>();
    }

    public int getValue() {
        return value;
    }

    public void setValue(int value) {
        this.value = value;
    }

    public int getX() {
        return x;
    }

    public void setX(int x) {
        this.x = x;
    }

    public int getY() {
        return y;
    }

    public void setY(int y) {
        this.y = y;
    }

    public Spot getPrevious() {
        return previous;
    }

    public void setPrevious(Spot previous) {
        this.previous = previous;
    }

    public ArrayList<Spot> getNeighbors() {
        return neighbors;
    }

    public void setNeighbors(ArrayList<Spot> neighbors) {
        this.neighbors = neighbors;
    }

    public double getG() {
        return g;
    }

    public void setG(double g) {
        this.g = g;
    }

    public double getH() {
        return h;
    }

    public void setH(double h) {
        this.h = h;
    }

    public double getF() {
        return f;
    }

    public void setF(double f) {
        this.f = f;
    }

    private int value;
    private int x;
    private int y;
    private Spot previous;
    private double g;
    private double h;
    private double f;
    private ArrayList<Spot> neighbors;
}

File: C#/A_Star/A_Star/A_Star/Spot.cs
namespace Main.A_Star;

public class Spot
{
    public int value
    {
        get => _value;
        set => _value = value;
    }
    public double f
    {
        get => _f;
        set => _f = value;
    }
    public double g
    {
        get => _g;
        set => _g = value;
    }
    public double h
    {
        get => _h;
        set => _h = value;
    }
    public Spot previous
    {
        get => _previous;
        set => _previous = value;
    }
    public List<Spot> neighbors
    {
        get => _neighbors;
        set => _neighbors = value;
    }

    public bool PositionEquals(Spot other)
    {
        return x == other.x && y == other.y;
    }
    
    public override bool Equals(object? obj)
    {
        Spot spot = (Spot)obj;
        return spot.x == this.x && spot.y == this.y;
    }

    private int _value;
    public int x;
    public int y;
    private double _f;
    private double _g;
    private double _h;
    private Spot _previous;
    private List<Spot> _neighbors;

    public Spot(int value, int x, int y)
    {
        this._value = value;
        this.x = x;
        this.y = y;
        this._f = 0;
        this._g = 0;
        this._h = 0;
        this._previous = null;
        this._neighbors = new List<Spot>();
    }
    
}